kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -294,4 +299,12 @@ public void processCallSummaryRecord(MotechEvent event) {
             deactivateSubscription(subscription, callRetry);
         }
     }
+
+    private void resetWelcomeFlagInSubscription(Subscription subscription) {
+
+        if (subscription.getNeedsWelcomeMessage() == true) {
```

Comment:
We can at least save a DB call with just one check for each KK subscription created via IVR.